xdem/workflows/workflows.py

```diff
@@ -0,0 +1,196 @@
+        self.level = self.config["outputs"]["level"]
+
+        self.outputs_folder = Path(self.config["outputs"]["path"])
+        self.outputs_folder.mkdir(parents=True, exist_ok=True)
+
+        for folder in ["png", "raster", "csv"]:
+            Path(self.outputs_folder / folder).mkdir(parents=True, exist_ok=True)
+
+        self.dico_to_show = [
+            ("Information about inputs", self.config["inputs"]),
+        ]
+
+    class NoAliasDumper(SafeDumper):  # type: ignore
+        """
+        NoAliasDumper to avoid id in YAML file
+        """
+
+        def ignore_aliases(self, data: Any) -> bool:
+            """
+            avoid id in YAML file
+            """
+            return True
+
+    def load_config(self) -> Dict[str, Any]:
+        """
+        Load a configuration file
+        :return: Configuration dictionary
+        """
+        if not os.path.exists(self.config_path):
+            raise FileNotFoundError(f"File not found : {self.config_path}")
+        with open(self.config_path) as f:
+            return yaml.safe_load(f)
+
+    def generate_graph(self, dem: RasterType, title: str, mask_path: str = None, **kwargs: Any) -> None:
+        """
+        Generate plot from a DEM
+        :param dem: Digital Elevation model
+        :param title: title of graph
+        :param mask_path: Path to mask
+        :return: None
+        """
+
+        plot_title = title.replace("_", " ")
+
+        if mask_path is None:
+            dem.plot(title=plot_title, **kwargs)
+            plt.savefig(self.outputs_folder / "png" / f"{title}.png")
+            plt.close()
+        else:
+            mask = gu.Vector(mask_path)
+            mask = mask.crop(dem)
+            dem.plot(title=plot_title, **kwargs)
+            mask.plot(dem, ec="k", fc="none")
+            plt.savefig(self.outputs_folder / "png" / f"{title}.png")
+            plt.close()
+
+    def floats_process(
+        self, dict_with_floats: Dict[str, Any] | InputCoregDict | OutputCoregDict | Any
+    ) -> Dict[str, Any]:  # type: ignore
+        """
+        Allows rounding all floats present in a dictionary to two decimal places
+        :param dict_with_floats: Dictionary with float
+        :return: Dictionary with floats
+        """
+        if isinstance(dict_with_floats, dict):
+            return {k: self.floats_process(v) for k, v in dict_with_floats.items()}
+        elif isinstance(dict_with_floats, list):
+            return [self.floats_process(elem) for elem in dict_with_floats]  # type: ignore
+        elif isinstance(dict_with_floats, tuple):
+            return tuple(self.floats_process(elem) for elem in dict_with_floats)  # type: ignore
+        elif isinstance(dict_with_floats, (float, np.floating)):
+            return round(float(dict_with_floats), 2)  # type: ignore
+        else:
+            return dict_with_floats
+
+    @staticmethod
+    def generate_dem(config_dem: Dict[str, Any] | None) -> tuple[DEM, Mask, str | None]:
```

Comment:
Maybe `load_dem` for the function name?
To answer your question in the PR description: I think we had a function that did this (especially for raising proper errors), but I'm not sure we still have it. See https://github.com/GlacioHack/geoutils/issues/595. I don't think this is major given that errors are raised mostly by Rasterio/Pyogrio directly.

We might also rename `nodata` into `force_nodata` here to be clearer (nodata is normally defined in any raster's metadata, so a user using this argument is overridding it).

Reply:
We use `force_source_nodata` in several places already: 
https://github.com/GlacioHack/geoutils/blob/e79a1ff8948709fda445835f30cf7dae649e046b/geoutils/raster/raster.py#L2545